hs_core/hydroshare/hs_bagit.py

```diff
@@ -138,6 +138,16 @@ def create_bag_files(resource):
     for f in resFiles:
         a.add_resource(f)
 
+    # in current implementation, collection has no resource files
+    if resource.resource_type == "CollectionResource" and resource.metadata.collection:
+        for contained_res in resource.metadata.collection.resources.all():
+            contained_res_id = contained_res.short_id
+            landing_page = '{hs_url}/resource/{res_id}/'.format(hs_url=current_site_url, res_id=contained_res_id)
```

Comment:
@pkdash that is good question. I am not sure we should save contained res landing page urls or bag urls in resourcemap.xml, or both? or just res id? Let's defer to @horsburgh 

Regarding your suggestion on re-organizing the code, actually I did what you said in first draft, and then changed it to current implementation. I was thinking maybe one day we will decide to allow collections hold both files and other resources, then current implementation is compatible with that...what do you think? thanks